app/controllers/placements/providers/placements_controller.rb

```diff
@@ -99,4 +102,20 @@ def compact_school_attribute_values(attribute)
     all_schools.where.not(attribute => nil)
                .distinct(attribute).order(attribute).pluck(attribute)
   end
+
+  def filter_subjects_by_phase
+    if filter_form.primary_selected? && !filter_form.secondary_selected?
+      @subjects = @subjects.where(subject_area: "primary")
+    elsif !filter_form.primary_selected? && filter_form.secondary_selected?
+      @subjects = @subjects.where(subject_area: "secondary")
+    end
+  end
```

Comment:
I would consider using this to replace the assignment on line 8. That was the assignment to `@subject` all happens here, as a single source of truth. Something like:
```
@subjects = filter_subjects_by_phase # Line 8

...

def filter_subjects_by_phase
  if filter_form.primary_selected? && !filter_form.secondary_selected?
    Subject.primary
  elsif !filter_form.primary_selected? && filter_form.secondary_selected?
    Subject.secondary
  else
    Subject
  end.order_by_name.select(:id, :name)
end
```
You might notice I've used the enum scopes for `subject_area` to get subjects `primary` and `secondary`. Just a little something nice you can do with enums.

Reply:
enum scopes :sparkles: